Database errors

justinThompson opened the pull request:
- api now detects if it cannot connect to the database
- displays an error message if it cannot connect

Commit:
- Initial commit of the Angular App
- Initial commit of the Node JS Server

Commit:
- updated readme credits to point to relative modules

Commit:
- added 500 response if cannot connect to database
- added message to the app to display error

sjcorbett commented on server/api.js:
```diff
@@ -1,50 +1,70 @@
     }
 };
+
+function canConnectToDatabases(err, res) {
```
Minor structural point but I think it'll lend clarity to the calling code: can you change each caller to 
```
if (!err) {
  con.query(...)
} else {
  sendError(res);
}
```
Where `sendError` is this method altered to just call `res.status(500).send(..)`.

Commit:
- updated database timeout to 5 seconds
- changed the root of the app to /
- updated structure of database connection errors
